IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m

```diff
@@ -130,38 +168,44 @@ - (BOOL)updateSignInStateForHomeAccountId:(NSString *)homeAccountId
         return NO;
     }
     
-    MSIDAccountMetadataCacheItem *accountMetadataItem;
-    MSIDAccountMetadataCacheKey *key = [[MSIDAccountMetadataCacheKey alloc] initWitHomeAccountId:homeAccountId clientId:clientId];
+    NSError *localError;
+    MSIDAccountMetadataCacheItem *cacheItem;
+    MSIDAccountMetadataCacheKey *key = [[MSIDAccountMetadataCacheKey alloc] initWithClientId:clientId];
+    cacheItem = [_metadataCache accountMetadataCacheItemWithKey:key context:context error:&localError];
+    if (localError)
+    {
+        if (error) *error = localError;
+        return NO;
+    }
+    
+    if (!cacheItem)
+    {
+        cacheItem = [[MSIDAccountMetadataCacheItem alloc] initWithClientId:clientId];
```

Comment:
nit: code to retrieve MSIDAccountMetadataCacheItem seems quite similar between URL update and sign-in state. Can we combine with a helper function?

Reply:
addressed :)